react/features/filmstrip/subscriber.web.js

```diff
@@ -153,3 +160,92 @@ StateListenerRegistry.register(
             store.dispatch(setTileViewDimensions(gridDimensions));
         }
     });
+
+/**
+ * Listens for changes to the screensharing status of the remote participants to recompute the reordered list of the
+ * remote endpoints.
+ */
+StateListenerRegistry.register(
+    /* selector */ state => state['features/video-layout'].remoteScreenShares,
+    /* listener */ (remoteScreenShares, store) => _reorderScreenshares(store));
+
+StateListenerRegistry.register(
+    /* selector */ state => state['features/base/participants'].dominantSpeaker,
+    /* listener */ (dominantSpeaker, store) => _reorderDominantSpeakers(store));
+
+/**
+ * Private helper function that reorders the remote participants based on dominant speaker changes.
+ *
+ * @param {*} store - The redux store.
+ * @returns {void}
+ * @private
+ */
+function _reorderDominantSpeakers(store) {
+    const state = store.getState();
+    const { dominantSpeaker, local, previousSpeakers, remote } = state['features/base/participants'];
+
+    if (!remote) {
+        return;
+    }
+    const { remoteParticipants, visibleRemoteParticipants } = state['features/filmstrip'];
+    const reorderedParticipants = [ ...remoteParticipants ];
+    const speakersList = [];
+
+    dominantSpeaker && speakersList.push(dominantSpeaker);
+    previousSpeakers?.length && speakersList.push(...previousSpeakers);
+
+    // Move the speakers up the list only if both the below conditions are met.
+    // 1. They are not the local participant.
+    // 2. They are currently not visible in the filmstrip.
+    for (const speaker of speakersList.reverse()) {
```

Comment:
I think when somebody leaves the `previousSpeakers` array won't be placed in the correct alphabetical order place but it will be left just after the new `previousSpeakers` participants. Am I correct?